src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -93,6 +94,12 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
             return;
         }
 
+        Pattern usernamePattern = Pattern.compile("[$&+,:;=\\\\?@#|/'<>.^*()%!-]");
```

Comment:
I think ``.`` should be allowed

Reply:
Permitted `.`. Do we also allow `@` in username?

Reply:
`@` shouldn't be allowed and only one `.` should be allowed. Thoughts? @cwperks @peternied @davidlago

Reply:
Do we also allow `-`, `_` and `@` along with `.`? Please suggest.
In the current fix I have permitted the four special characters above.